refactor: deprecate `SemverConstraint` field and introduce generic `constraint` field (#4541)
akuity/kargo: docs/docs/50-user-guide/20-how-to-guides/30-working-with-warehouses.md

strategies are:

- `SemVer`: Selects the image with the tag that best matches a semantic

>   versioning constraint specified by the `constraint` field. With no

Reviewer: ```suggestion
  versioning constraint specified by the `constraint` field. If no such
```

>   versioning constraint specified by the `constraint` field. With no
>   constraint specified, the strategy simply selects the image with the

Reviewer: ```suggestion
  versioning constraint specified by the `constraint` field. If no such
  constraint is specified, the strategy simply selects the image with the
```

  semantically greatest tag. All tags that are not valid semantic versions are
  ignored.

    :::note
    Starting Kargo v1.7, the `semverConstraint` field is deprecated for
    container image subscriptions. It is recommended to use the `constraint`
    field instead. When using the `SemVer` strategy, `constraint` should be a
    valid semantic version range.

>     :::

Author: PTAL now.

>     :::

Author: sorry, i missed modifying the existing paragraph. Can you please review now?

          allowTags: ^nightly-\d{8}$
    ```

- `Digest`: This selects the image _currently_ referenced by some "mutable tag"
   (such as `latest`) specified by the `constraint` field.

> 

Reviewer: ```suggestion
```

> - `Digest`: This selects the image _currently_ referenced by some "mutable tag,"
>    such as `latest`.

Reviewer: ```suggestion
- `Digest`: This selects the image _currently_ referenced by some "mutable tag"
   (such as `latest`) specified by the `constraint` field.
```

>     The mutable tag name must be specified using the `constraint` field. This field
>     should be set to a valid tag name (e.g., "latest"). The selected image will
>     track the digest associated with that tag, which may change over time as new
>     images are pushed.

Reviewer: ```suggestion
```

    :::note

>     Prior to Kargo v1.7, the `semverConstraint` field was used to
>     specify the mutable tag name. This field is now deprecated
>     and will be removed in v1.9.0. Users should use the `constraint` field
>     instead.

Reviewer: ```suggestion
    If the `constraint` field is empty, the name of the mutable tag will
    (unintuitively) be determined by the value of the deprecated
    `semverConstraint` field. The `semverConstraint` field will be removed in
    v1.9.0.
```

    __Unintuitively, the mutable tag name must be specified using the
    `semverConstraint` field.__ Importantly, the _digest_ will change every time
    the tag is updated. Since the `semverConstraint` field is deprecated
    starting v1.7, it is recommended to use the `constraint` field that should

>     be set to a valid mutable tag name.

Author: Please see now.
